src/test/java/me/moodcat/soundcloud/SoundCloudExtractTest.java

```diff
@@ -0,0 +1,64 @@
+package me.moodcat.soundcloud;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Created by jaap on 4/28/15.
+ */
+public class SoundCloudExtractTest {
+
+    private static final String COOL_SONG = "https://soundcloud.com/pegboardnerds/"
+            + "who-the-fuck-is-paul-mccartney-kanye-west-x-jennifer-lawrence-x-pegboard-nerds";
+
+    private static final String SONG2_ARTIST = "katfyr";
+
+    private static final String SONG2_TITLE_ID = "binary-original-mix";
+
+    private static final String SONG2_TITLE = "Binary (Original Mix)";
+
+    private static final String SONG2_INFO_URL = "https://api.soundcloud.com/resolve.json"
+            + "?url=https://soundcloud.com/katfyr/binary-original-mix&client_id=b45b1aa10f1ac2941910a7f0d10f8e28";
```

Comment:
These tests depend on the internet connection. It would be better to have a mocked URL object in order to test the actual implementation.
Then integrations tests can be used to test the link with the actual SoundCloud API.